packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx

```diff
@@ -0,0 +1,288 @@
+                        alt={localizedQRCodeAltText}
+                        src={qrCodeSrc}
+                      />
+                    </div>
+                    <FtlMsg id="inline-totp-setup-on-completion-description">
+                      <p className="text-sm mb-4">
+                        Once complete, it will begin generating security codes
+                        for you to enter.
+                      </p>
+                    </FtlMsg>
+                  </>
+                ) : (
+                  <>
+                    <FtlMsg
+                      id="inline-totp-setup-enter-key-or-use-qr-instructions"
+                      elems={{
+                        toggleToQRButton: (
+                          <button
+                            onClick={() => {
+                              setShowQR(true);
+                            }}
+                            className="link-blue inline"
+                          >
+                            Scan QR code instead?
+                          </button>
+                        ),
+                      }}
+                    >
+                      <p className="text-sm m-2">
+                        Type this secret key into your authentication app.{' '}
+                        <button
+                          onClick={() => {
+                            setShowQR(true);
+                          }}
+                          className="link-blue inline"
+                        >
+                          Scan QR code instead?
+                        </button>
+                      </p>
+                    </FtlMsg>
+                    <div className="qr-code-container">
+                      <div className="qr-code-text">{secret}</div>
+                    </div>
+                    <FtlMsg id="inline-totp-setup-on-completion-description">
+                      <p className="text-sm mb-4">
+                        Once complete, it will begin generating security codes
+                        for you to enter.
+                      </p>
+                    </FtlMsg>
+                  </>
+                )}
+                <FormVerifyCode
+                  viewName="inline_totp_setup"
+                  email={email}
+                  onSubmit={handleSubmit(onSubmit)}
+                  formAttributes={{
+                    inputLabelText: 'Security code',
+                    inputFtlId: 'inline-totp-setup-security-code-placeholder',
+                    pattern: 'd{6}',
+                    maxLength: 6,
+                    submitButtonText: 'Ready',
+                    submitButtonFtlId: 'inline-totp-setup-ready-button',
+                  }}
+                  code={totpCodeValue}
+                  setCode={setTotpCodeValue}
+                  codeErrorMessage={totpErrorMessage}
+                  setCodeErrorMessage={setTotpErrorMessage}
+                />
+                <button className="link-blue text-sm mt-4">
```

Comment:
If this just takes users "back" or sends them somewhere, it's probably a link and not a button? 🤔 Same with the other instance of this.

Speaking of other instance, this could also be a component within this component, not extracted out to another file but just to DRY it up:

```
  const cancelSetupButton = ( // or link?
    <button type="button" className="link-blue text-sm">
      <FtlMsg id="inline-totp-setup-cancel-setup-button">Cancel setup</FtlMsg>
    </button>
  );
```

Then you can use it like `{cancelSetupButton}`.

Reply:
@millsoper I just realized this is in the old code:

```
  // If the user cancels out of the flow, redirect back to the RP with
  // an error message that's easy to parse. The user will need to be shown
  // the 2FA-required instructions again.
  _cancel() {
    const err = AuthErrors.toError('TOTP_REQUIRED');
    this.redirectWithErrorCode(err);
  },
```

So eh, I'd say leave it as a button for now.

Reply:
I put this as a button because I can't have a link without an `href` and we don't know how this is going to function yet, but I'll add a note to the functionality ticket!

Reply:
I mostly called this out since for l10n, we're calling this a `-button` in the ID, and I'm unsure if we'd need a brand new ID if should be a link instead. I think this'll end up staying a button though, I didn't check the old code for this one until after I'd commented. 🤦‍♀️